test/framework/resourceversion_helpers.go

```diff
@@ -123,16 +123,13 @@ func getObjectsWithResourceVersion(ctx context.Context, proxy ClusterProxy, name
 	objects := map[string]client.Object{}
 	for _, node := range graph {
 		nodeNamespacedName := client.ObjectKey{Namespace: node.Object.Namespace, Name: node.Object.Name}
-		obj := &metav1.PartialObjectMetadata{
-			TypeMeta: metav1.TypeMeta{
-				APIVersion: node.Object.APIVersion,
-				Kind:       node.Object.Kind,
-			},
-		}
+		obj := &unstructured.Unstructured{}
+		obj.SetAPIVersion(node.Object.APIVersion)
+		obj.SetKind(node.Object.Kind)
```

Comment:
I always thought we're printing the entire object diff if this fails, but apparently only the diff of metadata :/

Example: https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/periodic-cluster-api-e2e-latestk8s-main/1946147853424398336

This should make it much easier to debug the e2e test failures